### PR TITLE
feat: limit CE runs list.

### PR DESCRIPTION
**Pull request by BenjaminMummery**
[ORQSDK-767]

# The problem

The public API for listing workflows has some filtering options including limit. `CERuntime.list_workflow_runs()` does not make use of the options.

# This PR's solution

`CERuntime.list_workflow_runs()` and the underlying driver should be updated to make use of the limit option.

# Checklist

_Check that this PR satisfies the following items:_

- [X] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [X] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

[ORQSDK-767]: https://zapatacomputing.atlassian.net/browse/ORQSDK-767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/orquestra/sdk/_base/_driver/_client.py [hunk 1]
```diff
@@ -370,6 +371,9 @@ def list_workflow_runs(
             workflow_def = self.get_workflow_def(r.definitionId)
             workflow_runs.append(r.to_ir(workflow_def))
 
+        if limit is not None:
+            workflow_runs = workflow_runs[-limit:]
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_driver/_client.py:375]: Is there a way we can use CE's API to avoid client-side filtering?

Comment (alexjuda) [reply src/orquestra/sdk/_base/_driver/_client.py:375]: FYI: the API spec is here: https://github.com/zapatacomputing/workflow-driver/blob/main/openapi/src/resources/workflow-runs.yaml

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 These parameters let you set the default imports for all tasks in given workflow.
 If a task defines its own imports (either source, dependencies, or both) - it will overwrite workflow defaults.
 - Allow single imports as `dependency_imports` in `@task` decorators.
+- CERuntime list_workflow_runs now applies the `limit` parameter.
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply CHANGELOG.md]: Updated in [fac4646](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/111/commits/fac4646462d98c84768d8ce42547925de3f5aa92), now reads 
```
Listing workflow runs from Compute Engine now allows an upper limit to the number of runs to be listed to be set via the `limit` keyword.
```

File: src/orquestra/sdk/_base/_driver/_client.py (hunk 1 above)
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/_driver/_client.py]: As of [d7092e1](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/111/commits/d7092e1838efbd4bc9432e549773e8f61a9510f9), requests the precise number of workflow runs as are needed.

**Review by jamesclark-Zapata — APPROVED** [review]

**Review by alexjuda — APPROVED** [review]

**Pull request merged**